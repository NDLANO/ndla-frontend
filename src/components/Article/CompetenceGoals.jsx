/**
 * Copyright (c) 2018-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree. *
 */

import React from 'react';
import { useQuery } from '@apollo/client';
import PropTypes from 'prop-types';
import { CompetenceGoalTab } from '@ndla/ui';
import { isValidElementType } from 'react-is';
import Spinner from '@ndla/ui/lib/Spinner';

import { useTranslation } from 'react-i18next';
import { competenceGoalsQuery } from '../../queries';
import handleError from '../../util/handleError';
import { ArticleShape, SubjectShape } from '../../shapes';

export function groupByCurriculums(competenceGoals, addUrl = false) {
  const searchUrl = '/search?grepCodes=';
  const curriculumsObject = competenceGoals.reduce((acc, goal) => {
    const curriculum = acc[goal.curriculum.id] || {
      id: goal.curriculum.id,
      title: `${goal.curriculum.title} (${goal.curriculum.id})`,
      elements: [],
    };
    const setTitle = goal.competenceGoalSet
      ? ` - ${goal.competenceGoalSet.title} (${goal.competenceGoalSet.id})`
      : '';

    return {
      ...acc,
      [goal.curriculum.id]: {
        ...curriculum,
        elements: [
          ...curriculum.elements,
          {
            id: goal.id,
            name: `${goal.name}${setTitle}`,
            text: goal.text,
            url: addUrl ? searchUrl + goal.id : '',
          },
        ],
      },
    };
  }, {});

  return Object.keys(curriculumsObject).map(key => curriculumsObject[key]);
}

const CompetenceGoals = ({
  article,
  subject,
  language,
  wrapperComponent: Component,
  wrapperComponentProps,
}) => {
  const codes = article.grepCodes;
  const nodeId = article.oldNdlaUrl?.split('/').pop();
  const lang =
    article.supportedLanguages.find(l => l === language) ||
    article.supportedLanguages[0];
    const {t} = useTranslation();
  const { error, data, loading } = useQuery(competenceGoalsQuery, {
    variables: { codes, nodeId, language: lang },
  });

  if (error) {
    handleError(error);
    return null;
  }

  if (loading) {
    return <Spinner />;
  }

  const { competenceGoals, coreElements } = data;
  const LK06Goals = groupByCurriculums(
    competenceGoals.filter(goal => goal.type === 'LK06'),
    false,
  );
  const LK20Goals = groupByCurriculums(
    competenceGoals.filter(goal => goal.type === 'LK20'),
    false,
  );
  const LK20Elements = groupByCurriculums(coreElements || [], false);

  const competenceGoalsList = [
    ...(LK20Goals?.length
      ? [
          {
            id: '1',
            title: t('competenceGoals.competenceTabLK20label'),
            type: 'competenceGoals',
            groupedCompetenceGoals: LK20Goals,
          },
        ]
      : []),
    ...(LK20Elements?.length
      ? [
          {
            id: '2',
            title: t('competenceGoals.competenceTabCorelabel'),
            type: 'coreElement',
            groupedCoreElementItems: LK20Elements,
          },
        ]
      : []),
    ...(LK06Goals?.length
      ? [
          {
            id: '3',
            title: t('competenceGoals.competenceTabLK06label'),
            type: 'competenceGoals',
            groupedCompetenceGoals: LK06Goals,
          },
        ]
      : []),
  ];

  return (
    <Component {...wrapperComponentProps}>
      <CompetenceGoalTab title={subject?.name} list={competenceGoalsList} />
    </Component>
  );
};

CompetenceGoals.propTypes = {
  article: ArticleShape,
  subject: SubjectShape,
  language: PropTypes.string,
  wrapperComponent: (props, propName) => {
    if (props[propName] && !isValidElementType(props[propName])) {
      return new Error(
        `Invalid prop 'component' supplied to 'CompetenceGoals': the prop is not a valid React component`,
      );
    }
    return null;
  },
  wrapperComponentProps: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
};

export default CompetenceGoals;
