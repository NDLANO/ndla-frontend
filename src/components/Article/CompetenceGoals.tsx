/**
 * Copyright (c) 2018-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree. *
 */

import React, { ComponentType } from 'react';
import { useQuery } from '@apollo/client';
import { CompetenceGoalTab } from '@ndla/ui';
import { useTranslation } from 'react-i18next';
import { isValidElementType } from 'react-is';
import Spinner from '@ndla/ui/lib/Spinner';

import { competenceGoalsQuery } from '../../queries';
import handleError from '../../util/handleError';
import {
  GQLArticle,
  GQLCompetenceGoal,
  GQLCoreElement,
} from '../../graphqlTypes';
import { CompetenceGoalsType } from '../../interfaces';

interface Props {
  article: GQLArticle;
  language: string;
  wrapperComponent: ComponentType;
  wrapperComponentProps: any;
}

// We swap 'title' for 'name' when we fetch CompetenceGoals from GraphQL
interface LocalGQLCompetenceGoal extends Omit<GQLCompetenceGoal, 'title'> {
  name: string;
}

// We do the same as above with 'name' and 'text'
interface LocalGQLCoreElement
  extends Omit<GQLCoreElement, 'title' | 'description'> {
  name: string;
  text?: string;
}

interface CompetenceGoalsProps {
  competenceGoals: LocalGQLCompetenceGoal[];
  coreElements: LocalGQLCoreElement[];
}

interface ElementType {
  id: string;
  title: string;
  type: 'competenceGoals' | 'coreElement';
  groupedCompetenceGoals?: CompetenceGoalType[];
  groupedCoreElementItems?: CoreElementType[];
}

interface CompetenceGoalType {
  title: string;
  elements: {
    id: string;
    title: string;
    goals: {
      text: string;
      url: string;
      type: CompetenceGoalsType;
    }[];
  }[];
}

interface CoreElementType {
  title: string;
  elements: {
    id: string;
    name: string;
    text: string;
    goals: {
      id: string;
      name: string;
    }[];
  }[];
}

const getUniqueCurriculums = (
  competenceGoals: (LocalGQLCompetenceGoal | LocalGQLCoreElement)[],
): (
  | LocalGQLCompetenceGoal['curriculum']
  | LocalGQLCoreElement['curriculum']
)[] => {
  const curriculums = competenceGoals
    .filter(e => e.curriculum?.id)
    .map(competenceGoal => competenceGoal.curriculum);
  return Object.values(
    curriculums.reduce(
      (acc, current) => ({
        ...acc,
        [current!.id]: current,
      }),
      {},
    ),
  );
};

const getUniqueCompetenceGoalSet = (
  competenceGoals: LocalGQLCompetenceGoal[],
  curriculumId: string,
): LocalGQLCompetenceGoal['competenceGoalSet'][] => {
  const competenceGoalSet = competenceGoals
    .filter(e => e.competenceGoalSet?.id)
    .filter(e => e.curriculum?.id === curriculumId)
    .map(competenceGoal => competenceGoal.competenceGoalSet);
  return Object.values(
    competenceGoalSet.reduce(
      (acc, current) => ({
        ...acc,
        [current!.id]: current,
      }),
      {},
    ),
  );
};

const getUniqueCompetenceGoals = (
  competenceGoals: LocalGQLCompetenceGoal[],
  competenceGoalSetId: string,
  addUrl: boolean,
  searchUrl: string,
  goalType: CompetenceGoalsType,
) => {
  return competenceGoals
    .filter(
      competenceGoal =>
        competenceGoal.competenceGoalSet?.id === competenceGoalSetId,
    )
    .map(competenceGoal => ({
      text: competenceGoal.name,
      url: addUrl ? searchUrl + competenceGoal.id : 'a',
      type: goalType,
    }));
};

const sortElementsById = (
  elements: ElementType['groupedCompetenceGoals'],
): ElementType['groupedCompetenceGoals'] =>
  elements!.map(e => ({
    ...e,
    elements: e.elements.sort((a: any, b: any) => {
      if (a.id! < b.id!) return -1;
      if (a.id! > b.id!) return 1;
      return 0;
    }),
  }));

export const groupCompetenceGoals = (
  competenceGoals: LocalGQLCompetenceGoal[],
  addUrl: boolean = false,
  goalType: CompetenceGoalsType,
): ElementType['groupedCompetenceGoals'] => {
  const searchUrl = '/search?grepCodes=';
  const curriculumElements = getUniqueCurriculums(competenceGoals).map(
    curriculum => ({
      title: `${curriculum?.title} (${curriculum?.id})`,
      elements: getUniqueCompetenceGoalSet(competenceGoals, curriculum!.id).map(
        competenceGoalSet => ({
          id: competenceGoalSet!.id,
          title: `${competenceGoalSet?.title} (${competenceGoalSet!.id})`,
          goals: getUniqueCompetenceGoals(
            competenceGoals,
            competenceGoalSet!.id,
            addUrl,
            searchUrl,
            goalType,
          ),
        }),
      ),
    }),
  );
  return sortElementsById(curriculumElements);
};

const groupCoreElements = (
  coreElements: LocalGQLCoreElement[],
): ElementType['groupedCoreElementItems'] => {
  return getUniqueCurriculums(coreElements).map(curriculum => ({
    title: `${curriculum?.title} (${curriculum!.id})`,
    elements: coreElements
      .filter(e => e.curriculum?.id === curriculum!.id)
      .map(coreElement => ({
        id: coreElement!.id,
        name: coreElement!.name,
        text: coreElement.text!,
        goals: [], // Currently (13.09.21) CoreElements does not have any goals
      })),
  }));
};

const CompetenceGoals = ({
  article,
  language,
  wrapperComponent: Component,
  wrapperComponentProps,
}: Props) => {
  const { t } = useTranslation();
  const codes = article.grepCodes;
  const nodeId = article.oldNdlaUrl?.split('/').pop();
  const lang =
    article.supportedLanguages?.find(l => l === language) ||
    article.supportedLanguages?.[0];
  const { error, data, loading } = useQuery(competenceGoalsQuery, {
    variables: { codes, nodeId, language: lang },
  });

  if (error) {
    handleError(error);
    return null;
  }

  if (loading) {
    return <Spinner />;
  }

  const { competenceGoals, coreElements }: CompetenceGoalsProps = data;
  const LK06Goals = groupCompetenceGoals(
    competenceGoals.filter(goal => goal.type === 'LK06'),
    true,
    'LK06',
  );
  const LK20Goals = groupCompetenceGoals(
    competenceGoals.filter(goal => goal.type === 'LK20'),
    true,
    'LK20',
  );
  const LK20Elements = groupCoreElements(coreElements || []);

  const CompetenceGoalsLK20Template: ElementType = {
    id: '1',
    title: t('competenceGoals.competenceTabLK20label'),
    type: 'competenceGoals',
    groupedCompetenceGoals: LK20Goals,
  };

  const CoreElementsTemplate: ElementType = {
    id: '2',
    title: t('competenceGoals.competenceTabCorelabel'),
    type: 'coreElement',
    groupedCoreElementItems: LK20Elements,
  };

  const CompetenceGoalsLK06Template: ElementType = {
    id: '3',
    title: t('competenceGoals.competenceTabLK06label'),
    type: 'competenceGoals',
    groupedCompetenceGoals: LK06Goals,
  };

  const competenceGoalsList: ElementType[] = [
    ...(LK20Goals?.length ? [CompetenceGoalsLK20Template] : []),
    ...(LK20Elements?.length ? [CoreElementsTemplate] : []),
    ...(LK06Goals?.length ? [CompetenceGoalsLK06Template] : []),
  ];

  return (
    <Component {...wrapperComponentProps}>
      <CompetenceGoalTab list={competenceGoalsList} />
    </Component>
  );
};

CompetenceGoals.propTypes = {
  wrapperComponent: (props: any, propName: string) => {
    if (props[propName] && !isValidElementType(props[propName])) {
      return new Error(
        `Invalid prop 'component' supplied to 'CompetenceGoals': the prop is not a valid React component`,
      );
    }
    return null;
  },
};

export default CompetenceGoals;
