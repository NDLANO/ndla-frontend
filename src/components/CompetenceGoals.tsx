/**
 * Copyright (c) 2018-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree. *
 */

import { ComponentType } from 'react';
import { useTranslation } from 'react-i18next';
import { CompetenceGoalTab } from '@ndla/ui';
import { competenceGoalsQuery } from '../queries';
import handleError from '../util/handleError';
import {
  GQLCompetenceGoal,
  GQLCompetenceGoalsQuery,
  GQLCoreElement,
} from '../graphqlTypes';
import { CompetenceGoalsType } from '../interfaces';
import { useGraphQuery } from '../util/runQueries';

interface Props {
  supportedLanguages?: string[];
  subjectId?: string;
  codes?: string[];
  nodeId?: string;
  wrapperComponent: ComponentType;
  wrapperComponentProps: any;
  isOembed?: boolean;
}

// We swap 'title' for 'name' when we fetch CompetenceGoals from GraphQL
interface LocalGQLCompetenceGoal extends Omit<GQLCompetenceGoal, 'title'> {
  name: string;
}

// We do the same as above with 'name' and 'text'
interface LocalGQLCoreElement
  extends Omit<GQLCoreElement, 'title' | 'description'> {
  name: string;
  text?: string;
}

interface ElementType {
  id: string;
  title: string;
  type: 'competenceGoals' | 'coreElement';
  groupedCompetenceGoals?: CompetenceGoalType[];
  groupedCoreElementItems?: CoreElementType[];
}

interface CompetenceGoalType {
  title: string;
  elements: {
    id: string;
    title: string;
    goals: {
      text: string;
      url: string;
      type: CompetenceGoalsType;
    }[];
  }[];
}

interface CoreElementType {
  title: string;
  elements: {
    id: string;
    name: string;
    text: string;
    goals: {
      id: string;
      name: string;
    }[];
  }[];
}

const getUniqueCurriculums = (
  competenceGoals: (LocalGQLCompetenceGoal | LocalGQLCoreElement)[],
): (
  | LocalGQLCompetenceGoal['curriculum']
  | LocalGQLCoreElement['curriculum']
)[] => {
  const curriculums = competenceGoals
    .filter(e => e.curriculum?.id)
    .map(competenceGoal => competenceGoal.curriculum);
  return Object.values(
    curriculums.reduce(
      (acc, current) => ({
        ...acc,
        [current!.id]: current,
      }),
      {},
    ),
  );
};

const getUniqueCompetenceGoalSet = (
  competenceGoals: LocalGQLCompetenceGoal[],
  curriculumId: string,
): LocalGQLCompetenceGoal['competenceGoalSet'][] => {
  const competenceGoalSet = competenceGoals
    .filter(e => e.competenceGoalSet?.id)
    .filter(e => e.curriculum?.id === curriculumId)
    .map(competenceGoal => competenceGoal.competenceGoalSet);
  return Object.values(
    competenceGoalSet.reduce(
      (acc, current) => ({
        ...acc,
        [current!.id]: current,
      }),
      {},
    ),
  );
};

const getUniqueCompetenceGoals = (
  competenceGoals: LocalGQLCompetenceGoal[],
  competenceGoalSetId: string,
  addUrl: boolean,
  searchUrl: string,
  goalType: CompetenceGoalsType,
) => {
  return competenceGoals
    .filter(
      competenceGoal =>
        competenceGoal.competenceGoalSet?.id === competenceGoalSetId,
    )
    .map(competenceGoal => ({
      text: competenceGoal.name,
      url: addUrl ? searchUrl + competenceGoal.id : '',
      type: goalType,
    }));
};

const sortElementsById = (
  elements: ElementType['groupedCompetenceGoals'],
): ElementType['groupedCompetenceGoals'] =>
  elements!.map(e => ({
    ...e,
    elements: e.elements.sort((a: any, b: any) => {
      if (a.id! < b.id!) return -1;
      if (a.id! > b.id!) return 1;
      return 0;
    }),
  }));

export const groupCompetenceGoals = (
  competenceGoals: LocalGQLCompetenceGoal[],
  addUrl: boolean = false,
  goalType: CompetenceGoalsType,
  subjectId?: string,
): ElementType['groupedCompetenceGoals'] => {
  const searchUrl = subjectId
    ? `/search?subjects=${subjectId}&grepCodes=`
    : '/search?grepCodes=';
  const curriculumElements = getUniqueCurriculums(competenceGoals).map(
    curriculum => ({
      title: `${curriculum?.title} (${curriculum?.id})`,
      elements: getUniqueCompetenceGoalSet(competenceGoals, curriculum!.id).map(
        competenceGoalSet => ({
          id: competenceGoalSet!.id,
          title: `${competenceGoalSet?.title} (${competenceGoalSet!.id})`,
          goals: getUniqueCompetenceGoals(
            competenceGoals,
            competenceGoalSet!.id,
            addUrl,
            searchUrl,
            goalType,
          ),
        }),
      ),
    }),
  );
  return sortElementsById(curriculumElements);
};

const groupCoreElements = (
  coreElements: LocalGQLCoreElement[],
): ElementType['groupedCoreElementItems'] => {
  return getUniqueCurriculums(coreElements).map(curriculum => ({
    title: `${curriculum?.title} (${curriculum!.id})`,
    elements: coreElements
      .filter(e => e.curriculum?.id === curriculum!.id)
      .map(coreElement => ({
        id: coreElement!.id,
        name: coreElement!.name,
        text: coreElement.text!,
        goals: [], // Currently (13.09.21) CoreElements does not have any goals
      })),
  }));
};

const CompetenceGoals = ({
  codes,
  nodeId,
  subjectId,
  wrapperComponent: Component,
  wrapperComponentProps,
  supportedLanguages,
  isOembed,
}: Props) => {
  const { t, i18n } = useTranslation();
  const language =
    supportedLanguages?.find(l => l === i18n.language) ||
    supportedLanguages?.[0] ||
    i18n.language;

  const { error, data } = useGraphQuery<GQLCompetenceGoalsQuery>(
    competenceGoalsQuery,
    {
      variables: { codes, nodeId, language },
    },
  );

  if (error) {
    handleError(error);
    return null;
  }

  if (!data) return null;

  const { competenceGoals, coreElements } = data;
  const LK06Goals = groupCompetenceGoals(
    competenceGoals?.filter(goal => goal.type === 'LK06') ?? [],
    false,
    'LK06',
  );
  const LK20Goals = groupCompetenceGoals(
    competenceGoals?.filter(goal => goal.type === 'LK20') ?? [],
    true,
    'LK20',
    subjectId,
  );
  const LK20Elements = groupCoreElements(coreElements || []);

  const CompetenceGoalsLK20Template: ElementType = {
    id: '1',
    title: t('competenceGoals.competenceTabLK20label'),
    type: 'competenceGoals',
    groupedCompetenceGoals: LK20Goals,
  };

  const CoreElementsTemplate: ElementType = {
    id: '2',
    title: t('competenceGoals.competenceTabCorelabel'),
    type: 'coreElement',
    groupedCoreElementItems: LK20Elements,
  };

  const CompetenceGoalsLK06Template: ElementType = {
    id: '3',
    title: t('competenceGoals.competenceTabLK06label'),
    type: 'competenceGoals',
    groupedCompetenceGoals: LK06Goals,
  };

  const competenceGoalsList: ElementType[] = [
    ...(LK20Goals?.length ? [CompetenceGoalsLK20Template] : []),
    ...(LK20Elements?.length ? [CoreElementsTemplate] : []),
    ...(LK06Goals?.length ? [CompetenceGoalsLK06Template] : []),
  ];

  return (
    <Component {...wrapperComponentProps}>
      <CompetenceGoalTab list={competenceGoalsList} isOembed={isOembed} />
    </Component>
  );
};

export default CompetenceGoals;
