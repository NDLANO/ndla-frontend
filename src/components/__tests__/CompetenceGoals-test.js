/**
 * Copyright (c) 2018-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { groupCompetenceGoals } from '../CompetenceGoals';

const competenceGoals = [
  {
    id: 'KM2648',
    name: 'carry out basic first aid (KM2648)',
    type: 'LK20',
    curriculum: {
      id: 'HSF01-03',
      title: 'Curriculum vg1 healthcare, child and youth development',
    },
    competenceGoalSet: {
      id: 'KV244',
      title: 'Competence aims and assessment health promotion',
    },
  },
  {
    id: 'KM2647',
    name: 'discuss and give examples of what each person and society can do to improve their own health and public health (KM2647)',
    type: 'LK20',
    curriculum: {
      id: 'HSF01-03',
      title: 'Curriculum vg1 healthcare, child and youth development',
    },
    competenceGoalSet: {
      id: 'KV244',
      title: 'Competence aims and assessment health promotion',
    },
  },
  {
    id: 'KM1232',
    name: 'explain present-day changes in spoken Norwegian and reflect on relationships between language, culture and identity (KM1232)',
    type: 'LK20',
    curriculum: {
      id: 'NOR01-06',
      title: 'Curriculum for Norwegian',
    },
    competenceGoalSet: {
      id: 'KV115',
      title: 'Competence aims and assessment Vg3 programmes for general studies',
    },
  },
];

test('That groupByCurriculums groups competenceGoals by curriculum ', () => {
  const grouped = groupCompetenceGoals(competenceGoals, false, 'LK20');
  expect(grouped).toMatchSnapshot();
});

test('That addUrl param adds url to element ', () => {
  const grouped = groupCompetenceGoals(competenceGoals, true, 'LK20');
  expect(grouped).toMatchSnapshot();
});

test('That addUrl and subjectId params adds extended url to element ', () => {
  const grouped = groupCompetenceGoals(competenceGoals, true, 'LK20', 'urn:subject:20');
  expect(grouped).toMatchSnapshot();
});
