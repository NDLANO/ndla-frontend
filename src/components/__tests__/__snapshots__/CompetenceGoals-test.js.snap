// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`That addUrl and subjectId params adds extended url to element  1`] = `
Array [
  Object {
    "elements": Array [
      Object {
        "goals": Array [
          Object {
            "text": "carry out basic first aid (KM2648)",
            "type": "LK20",
            "url": "/search?subjects=urn:subject:20&grepCodes=KM2648",
          },
          Object {
            "text": "discuss and give examples of what each person and society can do to improve their own health and public health (KM2647)",
            "type": "LK20",
            "url": "/search?subjects=urn:subject:20&grepCodes=KM2647",
          },
        ],
        "id": "KV244",
        "title": "Competence aims and assessment health promotion (KV244)",
      },
    ],
    "title": "Curriculum vg1 healthcare, child and youth development (HSF01-03)",
  },
  Object {
    "elements": Array [
      Object {
        "goals": Array [
          Object {
            "text": "explain present-day changes in spoken Norwegian and reflect on relationships between language, culture and identity (KM1232)",
            "type": "LK20",
            "url": "/search?subjects=urn:subject:20&grepCodes=KM1232",
          },
        ],
        "id": "KV115",
        "title": "Competence aims and assessment Vg3 programmes for general studies (KV115)",
      },
    ],
    "title": "Curriculum for Norwegian (NOR01-06)",
  },
]
`;

exports[`That addUrl param adds url to element  1`] = `
Array [
  Object {
    "elements": Array [
      Object {
        "goals": Array [
          Object {
            "text": "carry out basic first aid (KM2648)",
            "type": "LK20",
            "url": "/search?grepCodes=KM2648",
          },
          Object {
            "text": "discuss and give examples of what each person and society can do to improve their own health and public health (KM2647)",
            "type": "LK20",
            "url": "/search?grepCodes=KM2647",
          },
        ],
        "id": "KV244",
        "title": "Competence aims and assessment health promotion (KV244)",
      },
    ],
    "title": "Curriculum vg1 healthcare, child and youth development (HSF01-03)",
  },
  Object {
    "elements": Array [
      Object {
        "goals": Array [
          Object {
            "text": "explain present-day changes in spoken Norwegian and reflect on relationships between language, culture and identity (KM1232)",
            "type": "LK20",
            "url": "/search?grepCodes=KM1232",
          },
        ],
        "id": "KV115",
        "title": "Competence aims and assessment Vg3 programmes for general studies (KV115)",
      },
    ],
    "title": "Curriculum for Norwegian (NOR01-06)",
  },
]
`;

exports[`That groupByCurriculums groups competenceGoals by curriculum  1`] = `
Array [
  Object {
    "elements": Array [
      Object {
        "goals": Array [
          Object {
            "text": "carry out basic first aid (KM2648)",
            "type": "LK20",
            "url": "",
          },
          Object {
            "text": "discuss and give examples of what each person and society can do to improve their own health and public health (KM2647)",
            "type": "LK20",
            "url": "",
          },
        ],
        "id": "KV244",
        "title": "Competence aims and assessment health promotion (KV244)",
      },
    ],
    "title": "Curriculum vg1 healthcare, child and youth development (HSF01-03)",
  },
  Object {
    "elements": Array [
      Object {
        "goals": Array [
          Object {
            "text": "explain present-day changes in spoken Norwegian and reflect on relationships between language, culture and identity (KM1232)",
            "type": "LK20",
            "url": "",
          },
        ],
        "id": "KV115",
        "title": "Competence aims and assessment Vg3 programmes for general studies (KV115)",
      },
    ],
    "title": "Curriculum for Norwegian (NOR01-06)",
  },
]
`;
